server/src/test/java/org/elasticsearch/health/node/LocalHealthMonitorTests.java

```diff
@@ -241,8 +255,135 @@ public void testEnablingAndDisabling() throws Exception {
+        assertBusy(() -> assertThat(requestCounter.get(), equalTo(2)));
+    }
+
+    /**
+     * The aim of this test is to rapidly fire off a series of state changes and make sure that the health node in the last cluster
+     * state actually gets the health info.
+     */
+    public void testRapidStateChanges() throws Exception {
+        ClusterState state = ClusterStateCreationUtils.state(node, node, node, new DiscoveryNode[] { node, frozenNode })
+            .copyAndUpdate(b -> b.putCustom(HealthMetadata.TYPE, healthMetadata));
+        doReturn(state).when(clusterService).state();
+
+        // Keep track of the "current" health node.
+        var currentHealthNode = new AtomicReference<>(node);
+        // Keep a list of all the health nodes that have received a request.
+        var updatedHealthNodes = new ArrayList<DiscoveryNode>();
+        doAnswer(invocation -> {
+            var diskHealthInfo = ((UpdateHealthInfoCacheAction.Request) invocation.getArgument(1)).getDiskHealthInfo();
+            assertThat(diskHealthInfo, equalTo(GREEN));
+            ActionListener<AcknowledgedResponse> listener = invocation.getArgument(2);
+            listener.onResponse(null);
+            updatedHealthNodes.add(currentHealthNode.get());
+            return null;
+        }).when(client).execute(any(), any(), any());
+
+        localHealthMonitor.setMonitorInterval(TimeValue.timeValueMillis(0));
+        localHealthMonitor.clusterChanged(new ClusterChangedEvent("start-up", state, ClusterState.EMPTY_STATE));
+
+        int count = randomIntBetween(10, 20);
+        for (int i = 0; i < count; i++) {
+            var previous = state;
+            state = mutateState(previous);
+            currentHealthNode.set(HealthNode.findHealthNode(state));
+            localHealthMonitor.clusterChanged(new ClusterChangedEvent("switch", state, previous));
+        }
+
+        var lastHealthNode = DiscoveryNodeUtils.create("health-node", "health-node");
+        var previous = state;
+        state = ClusterStateCreationUtils.state(
+            node,
+            previous.nodes().getMasterNode(),
+            lastHealthNode,
+            new DiscoveryNode[] { node, frozenNode, lastHealthNode }
+        ).copyAndUpdate(b -> b.putCustom(HealthMetadata.TYPE, healthMetadata));
+        currentHealthNode.set(lastHealthNode);
+        localHealthMonitor.clusterChanged(new ClusterChangedEvent("switch", state, previous));
+
+        assertBusy(() -> assertTrue(updatedHealthNodes.contains(lastHealthNode)));
+    }
+
+    private ClusterState mutateState(ClusterState previous) {
+        var masterNode = previous.nodes().getMasterNode();
+        var healthNode = HealthNode.findHealthNode(previous);
+        switch (randomInt(1)) {
+            case 0 -> masterNode = randomValueOtherThan(masterNode, () -> randomFrom(node, frozenNode));
+            case 1 -> healthNode = randomValueOtherThan(healthNode, () -> randomFrom(node, frozenNode));
```

Comment:
Should we have a few more nodes to rotate/combine here? 5 perhaps? (so we get a new health/master combo more often?)

Reply:
See [4d8c947](https://github.com/elastic/elasticsearch/pull/105674/commits/4d8c947278bac2396b7a395adc7b86f00482416c).